Modules/Office365DSC/DSCResources/MSFT_TeamsGuestSettings/MSFT_TeamsGuestSettings.psm1

```diff
@@ -0,0 +1,183 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TeamName,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowCreateUpdateChannels,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowDeleteChannels,
+
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Test-TeamsServiceConnection -GlobalAdminAccount $GlobalAdminAccount
+
+    $nullReturn = @{
+        TeamName                  = $TeamName
+        AllowCreateUpdateChannels = $null
+        AllowDeleteChannels        = $null
+        Ensure                    = "Absent"
+        GlobalAdminAccount        = $GlobalAdminAccount
+    }
+
+    Write-Verbose -Message "Getting Team guest settings for $TeamName"
+
+    $team = Get-Team |  Where-Object {$_.DisplayName -eq $TeamName}
+    if ($null -eq $team)
+    {
+        throw "Team with Name $TeamName doesnt exist in tenant"
+    }
+
+    $teamGuestSettings = Get-TeamGuestSettings -GroupId $team.GroupId -ErrorAction SilentlyContinue
+    if ($null -eq $teamGuestSettings)
+    {
+        Write-Verbose "The specified Team doesn't exist."
+        return $nullReturn
+    }
+
+    Write-Verbose "Team guest settings for AllowCreateUpdateChannels = $($teamGuestSettings.AllowCreateUpdateChannels)"
+    Write-Verbose "Team guest settings for AllowDeleteChannels = $($teamGuestSettings.AllowDeleteChannels)"
+
+
+    return @{
+        TeamName                  = $TeamName
+        AllowCreateUpdateChannels = $teamGuestSettings.AllowCreateUpdateChannels
+        AllowDeleteChannels       = $teamGuestSettings.AllowDeleteChannels
+        Ensure                    = "Present"
+        GlobalAdminAccount        = $GlobalAdminAccount
+    }
+
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $TeamName,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowCreateUpdateChannels,
+
+        [Parameter()]
+        [System.Boolean]
+        $AllowDeleteChannels,
+
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    if ('Absent' -eq $Ensure)
+    {
+        throw "This resource cannot delete Managed Properties. Please make sure you set its Ensure value to Present."
```

Comment:
Make sure text reflects the appropriate text. Something like "This resource doesn't handle Ensure = Absent. Please make sure its Ensure value is set to Present" or something more appropriate

Reply:
done